File: demo/android/DemoRamsesAndroidModule/src/main/java/com/bmwgroup/ramses/UniformInput.java
//  -------------------------------------------------------------------------
//  Copyright (C) 2018 BMW Car IT GmbH
//  -------------------------------------------------------------------------
//  This Source Code Form is subject to the terms of the Mozilla Public
//  License, v. 2.0. If a copy of the MPL was not distributed with this
//  file, You can obtain one at https://mozilla.org/MPL/2.0/.
//  -------------------------------------------------------------------------

package com.bmwgroup.ramses;

public class UniformInput {
    public UniformInput(long handle)
    {
        m_nativeHandle = handle;
    }

    public void setValueFloat(float x)
    {
        setValueFloatNative(m_nativeHandle, x);
    }

    public void dispose()
    {
        disposeNative(m_nativeHandle);
    }

    private long m_nativeHandle;

    private native void setValueFloatNative(long handle, float x);
    private native void disposeNative(long handle);
}

File: src/client/impl/UniformInput.cpp
//  -------------------------------------------------------------------------
//  Copyright (C) 2014 BMW Car IT GmbH
//  -------------------------------------------------------------------------
//  This Source Code Form is subject to the terms of the Mozilla Public
//  License, v. 2.0. If a copy of the MPL was not distributed with this
//  file, You can obtain one at https://mozilla.org/MPL/2.0/.
//  -------------------------------------------------------------------------

#include "ramses/client/UniformInput.h"
#include "impl/EffectInputImpl.h"

namespace ramses
{
    UniformInput::UniformInput()
        : EffectInput{ std::make_unique<internal::EffectInputImpl>() }
    {
    }

    UniformInput::~UniformInput() = default;

    UniformInput::UniformInput(const UniformInput& other) = default;

    UniformInput::UniformInput(UniformInput&& other) noexcept = default;

    UniformInput& UniformInput::operator=(const UniformInput& other) = default;

    UniformInput& UniformInput::operator=(UniformInput&& other) noexcept = default;

    EEffectUniformSemantic UniformInput::getSemantics() const
    {
        return m_impl->getUniformSemantics();
    }

    size_t UniformInput::getElementCount() const
    {
        return m_impl->getElementCount();
    }
}
